test/fileurlreader.cpp

```diff
@@ -95,3 +96,77 @@ TEST_CASE("Preserves URLs as-is", "[FileUrlReader][issue926]")
 	REQUIRE(u.get_urls()[0] ==
 		R"_(exec:curl --silent https://feeds.metaebene.me/raumzeit/m4a  | sed 's#\(</guid>\|</id>\)#-M4A&#')_");
 }
+
+TEST_CASE("URL reader returns error message if file cannot be opened",
+	"[FileUrlReader]")
+{
+	const std::string testDataPath("data/test-urls.txt");
+
+	TestHelpers::TempFile urlsFile;
+	FileUrlReader u(urlsFile.get_path());
+
+	SECTION("reload() returns error message if file does not exist") {
+		const auto error_message = u.reload();
+		REQUIRE(error_message.has_value());
+
+		SECTION("the error message contains the filename") {
+			INFO("error_message: " + error_message.value());
+			REQUIRE(error_message.value().find(urlsFile.get_path()) != std::string::npos);
+		}
+	}
+
+	SECTION("write_config() works fine if file does not exist") {
+		const auto error_message = u.write_config();
+		REQUIRE(error_message.has_value() == false);
```

Comment:
```suggestion
		REQUIRE_FALSE(error_message.has_value());
```

Ditto in a few places below.

Reply:
Done, thanks